lib/jekyll-admin/server/page.rb

```diff
@@ -12,6 +12,11 @@ class Server < Sinatra::Base
       end
 
       put "/*?/?:path.:ext" do
+        unless html_content?
```

Comment:
I'd move this to a helper method, e.g., `ensure_html_content` to mirror the `ensure_document` and `ensure_directory` helpers.